frontendUi/src/main/java/com/virtuslab/gitmachete/frontend/ui/cell/BranchOrCommitCellRenderer.java

```diff
@@ -137,29 +139,38 @@ void customize(BranchOrCommitCell cell, boolean isSelected, boolean hasFocus, in
     }
 
     @UIEffect
-    private int calculateTextPadding(int maxPosition) {
+    @Positive
+    private int calculateTextPadding(@NonNegative int maxPosition) {
       int width = (maxPosition + 1) * PaintParameters.getNodeWidth(graphTable.getRowHeight());
-      return width + LabelPainter.RIGHT_PADDING.get();
+      int padding = width + LabelPainter.RIGHT_PADDING.get();
+      // Our assumption here comes from the fact that we expect positive row height of graph table
```

Comment:
Nitpicking (or maybe not): 

> we expect positive row height of graph table

Even though `graphTable.getRowHeight()` is positive, it still doesn't mean that `PaintParameters.getNodeWidth(graphTable.getRowHeight()` (and hence `width`) is positive; the current impl of `PaintParameters.getNodeWidth` does the following:

```java
  private static final int WIDTH_NODE = 15;
  // ...
  public static final int ROW_HEIGHT = 22;

  public static int getNodeWidth(int rowHeight) {
    return WIDTH_NODE * rowHeight / ROW_HEIGHT;
  }
```

so for positive `rowHeight == 1` it's still zero... maybe not very important but worth mentioning (esp. given that the constants in `PaintParameters` might change in future)

Reply:
Or maybe we can just rely on `LabelPainter.RIGHT_PADDING.get()` being **positive** and `width` being **non-negative**?

Reply:
comment updated